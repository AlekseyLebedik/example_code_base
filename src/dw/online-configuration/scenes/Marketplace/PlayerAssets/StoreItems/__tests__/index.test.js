import React from 'react';
import { shallow } from 'enzyme';

import { StoreItemsDisconnected } from '../index';

describe('Store Inventory', () => {
  const onSelectRarityFilter = jest.fn();
  const onInventoryFilterChange = jest.fn();
  const onSelectItems = jest.fn();
  const grantItems = jest.fn();
  const getStoreItems = jest.fn();

  const renderContainer = (customProps = {}) => {
    const propsConnected = {
      onSelectRarityFilter,
      onInventoryFilterChange,
      onSelectItems,
      grantItems,
      getStoreItems,
      selectedItems: [],
      missingItems: [],
      items: [],
      ...customProps,
    };

    return shallow(<StoreItemsDisconnected {...propsConnected} />);
  };

  describe('Connected Store Inventory', () => {
    it('renders default values', () => {
      const wrapper = renderContainer();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
