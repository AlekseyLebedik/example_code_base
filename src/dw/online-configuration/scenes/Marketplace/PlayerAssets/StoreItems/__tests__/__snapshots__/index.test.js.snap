// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store Inventory Connected Store Inventory renders default values 1`] = `
<InventoryItemsStateless
  actionsRender={[Function]}
  checked={false}
  classes={Object {}}
  columnsDefs={
    Array [
      Object {
        "field": "maxQuantity",
        "filter": "agNumberColumnFilter",
        "menuTabs": Array [
          "filterMenuTab",
          "columnsMenuTab",
        ],
      },
    ]
  }
  darkMode={true}
  externalActions={false}
  formatDate={[Function]}
  getStoreItems={[MockFunction]}
  grantItems={[MockFunction]}
  hasEditPermission={false}
  invalidItems={Array []}
  isLoading={false}
  items={Array []}
  missingItems={Array []}
  onInventoryFilterChange={[MockFunction]}
  onSelectItems={[MockFunction]}
  onSelectRarityFilter={[MockFunction]}
  searchInput=""
  selectedItems={Array []}
  storeLabel=""
  userId={null}
  validContext={null}
/>
`;
